File: CH2-modules/EX10-monthly-sales-tax/source-code/monthlySalesTax.js
const prompt = require("prompt-sync")();

function inputTotalSale(totalSale) {
  totalSale.value = Number(prompt("Total Sale: $"));
}

function processTax(totalSale, totalStateTax, totalCountryTax, totalSalesTax) {
  let stateTax = 0.04;
  let countryTax = 0.02;
  totalStateTax.value = totalSale.value * stateTax;
  totalCountryTax.value = totalSale.value * countryTax;
  totalSalesTax.value = totalStateTax.value + totalCountryTax.value;
}

function printTax(totalStateTax, totalCountryTax, totalSalesTax) {
  console.log("State Sales Tax   : $" + totalStateTax.value);
  console.log("Country Sales Tax : $" + totalCountryTax.value);
  console.log("Total Sales Tax   : $" + totalSalesTax.value);
}

function main() {
  let totalStateTax = {
    value: 0,
  };
  let totalCountryTax = {
    value: 0,
  };
  let totalSalesTax = {
    value: 0,
  };
  let totalSale = {
    value: 0,
  };

  inputTotalSale(totalSale);
  processTax(totalSale, totalStateTax, totalCountryTax, totalSalesTax);
  printTax(totalStateTax, totalCountryTax, totalSalesTax);
}

// entry point
main();

File: CH2-modules/EX9-monthly-sales-tax/source-code/monthlySalesTax.cpp
#include <iostream>
using namespace std;

void inputTotalSale(double &totalSale)
{
    cout << "Total Sale: $";
    cin >> totalSale;
}

void processTax(double &totalSale, double &totalStateTax, double &totalCountryTax, double &totalSalesTax)
{
    double stateTax = 0.04;
    double countryTax = 0.02;
    totalStateTax = totalSale * stateTax;
    totalCountryTax = totalSale * countryTax;
    totalSalesTax = totalStateTax + totalCountryTax;
}

void printTax(double &totalStateTax, double &totalCountryTax, double &totalSalesTax)
{
    cout << "State Sales Tax   : $" << totalStateTax << endl;
    cout << "Country Sales Tax : $" << totalCountryTax << endl;
    cout << "Total Sales Tax   : $" << totalSalesTax << endl;
}

int main()
{
    double totalStateTax;
    double totalCountryTax;
    double totalSalesTax;
    double totalSale;

    inputTotalSale(totalSale);
    processTax(totalSale, totalStateTax, totalCountryTax, totalSalesTax);
    printTax(totalStateTax, totalCountryTax, totalSalesTax);
}